app/webpacker/components/Delegates/DelegatesTable.jsx

```diff
@@ -8,44 +8,113 @@ import UserBadge from '../UserBadge';
 
 const dasherize = (string) => _.kebabCase(string);
 
+const SORT_TYPES = {
+  ASCENDING: 'ascending',
+  DESCENDING: 'descending',
+};
+
 export default function DelegatesTable({
-  delegates, isAdminMode, isAllLeadDelegates, isAllNonLeadDelegates,
+  delegates,
+  isAdminMode,
+  isAllLeadDelegates,
+  isAllNonLeadDelegates,
 }) {
-  const tableData = useMemo(() => delegates.filter(
-    (delegate) => delegate.metadata.status !== 'trainee_delegate' || isAdminMode,
-  ), [delegates, isAdminMode]);
+  const sortedDataReducer = (state, action) => {
+    switch (action.type) {
+      case 'CHANGE_SORT':
+        if (state.column === action.column) {
+          return {
+            ...state,
+            data: state.data.slice().reverse(),
+            direction:
+              state.direction === SORT_TYPES.ASCENDING
+                ? SORT_TYPES.DESCENDING
+                : SORT_TYPES.ASCENDING,
+          };
+        }
+        return {
+          column: action.column,
+          data: _.sortBy(state.data, [action.column]),
+          direction: SORT_TYPES.ASCENDING,
+        };
+      default:
+        throw new Error();
+    }
+  };
```

Comment:
It's best to define reducers outside of the component. (To ensure it's not accidentally using any state from the component, and so it's not re-computed every render.)